homeassistant/components/improv_ble/config_flow.py

```diff
@@ -170,6 +171,20 @@ async def async_step_start_improv(
         """
         # mypy is not aware that we can't get here without having these set already
         assert self._discovery_info is not None
+        self._discovery_info = async_last_service_info(
```

Comment:
Also store in a local variable so we don't need to access the instance attribute again on the next line.